src/Symfony/Component/Translation/Tests/Catalogue/TargetOperationTest.php

```diff
@@ -71,13 +71,45 @@ public function testGetResultWithMixedDomains()
     {
         $this->assertEquals(
             new MessageCatalogue('en', [
-                'messages+intl-icu' => ['a' => 'old_a'],
+                'messages' => ['a' => 'old_a'],
             ]),
             $this->createOperation(
                 new MessageCatalogue('en', ['messages' => ['a' => 'old_a']]),
                 new MessageCatalogue('en', ['messages+intl-icu' => ['a' => 'new_a']])
             )->getResult()
         );
+
+        $this->assertEquals(
+            new MessageCatalogue('en', [
+                'messages+intl-icu' => ['a' => 'old_a'],
+            ]),
+            $this->createOperation(
+                new MessageCatalogue('en', ['messages+intl-icu' => ['a' => 'old_a']]),
+                new MessageCatalogue('en', ['messages' => ['a' => 'new_a']])
+            )->getResult()
+        );
+
+        $this->assertEquals(
```

Comment:
This is testing the behavior when merging with new messages and mixed domains. Should this be moved into a separate function/test case? Or is it OK to have it all in the same function?